google/resource_sql_database_test.go

```diff
@@ -14,14 +14,27 @@ import (
 func TestAccGoogleSqlDatabase_basic(t *testing.T) {
 	var database sqladmin.Database
 
+	instance_name := acctest.RandString(10)
+	database_name := acctest.RandString(10)
+
 	resource.Test(t, resource.TestCase{
 		PreCheck:     func() { testAccPreCheck(t) },
 		Providers:    testAccProviders,
 		CheckDestroy: testAccGoogleSqlDatabaseInstanceDestroy,
 		Steps: []resource.TestStep{
 			resource.TestStep{
 				Config: fmt.Sprintf(
-					testGoogleSqlDatabase_basic, acctest.RandString(10), acctest.RandString(10)),
+					testGoogleSqlDatabase_basic, instance_name, database_name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGoogleSqlDatabaseExists(
+						"google_sql_database.database", &database),
+					testAccCheckGoogleSqlDatabaseEquals(
+						"google_sql_database.database", &database),
+				),
+			},
+			resource.TestStep{
```

Comment:
Let's split the charset and collation parts into their own test and leave `_basic` as a single-step test; that way, we can keep the testing for `Update` out of `_basic`.